### PR TITLE
[WIP] Add rust POC

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Move some of the functionalities to a rust folder.

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- The tests should pass as they are now - since that functionality is tested in Python

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by kamathhrishi — COMMENTED** [review]
It would have in the description of the motivation behind doing this. Further, the components we planned to move to rust.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,8 +39,7 @@ repos:
     rev: 1.7.0
     hooks:
       - id: bandit
-        args: ["--exclude", "tests"]
-
+        exclude: tests
```
Comment (arturomf94) [inline .pre-commit-config.yaml:42]: @gmuraru why was this changed?

**Comment by gmuraru** [comment]
Closing this for the moment

**Comment by arturomf94** [comment]
:(

**Pull request closed without merging**